### PR TITLE
Add additional quiet check in update-report

### PR DESCRIPTION
**Pull request by dkav**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [X] Have you successfully run `brew style` with your changes locally?
- [X] Have you successfully run `brew typecheck` with your changes locally?
- [X] Have you successfully run `brew tests` with your changes locally?

-----
An additional `--quiet` check that excludes reporting of tagged releases and relevant online links. Revision changes are still reported. Builds on https://github.com/Homebrew/brew/pull/10844.

### CODE REVIEW

**Comment by carlocab** [comment]
Thanks for your PR, @dkav.

I don't think it's a good idea to silence these messages with a `--quiet` flag.

One reason why these messages are printed out is that we've gotten a number of rather loud complaints about breaking changes and users being unaware of these. It's important that we try to make users more aware of breaking changes, and one way to try to do that is to make sure everyone sees these update messages.

New releases don't happen very often (the last two were an exception: they were tagged to ship security fixes), so I don't think it's that burdensome to not be able to silence these messages.

That said, if you still feel that strongly about this, it _might_ be worth pursuing a more targetted silencing of these messages, so that there's a way to get the really important updates to users when they do `brew update`.

**Comment by MikeMcQuaid** [comment]
> One reason why these messages are printed out is that we've gotten a number of rather loud complaints about breaking changes and users being unaware of these. It's important that we try to make users more aware of breaking changes, and one way to try to do that is to make sure everyone sees these update messages.

Or, at the very least, `--quiet` should silence announcing patch releases but not major/minor releases.

**Comment by dkav** [comment]
Thanks for the feedback. I amended the change so that `--quiet` silences patch release announcements only.

**Review by Rylan12 — CHANGES_REQUESTED** [review]
One other thing I'd like to see if this change is made is for the `Settings.write "latesttag", new_tag` call on [line 95](https://github.com/Homebrew/brew/blob/7e2ba95169593366565e84bce44da28a5aefe62c/Library/Homebrew/cmd/update-report.rb#L95) to be moved down to where the message is actually displayed.

Calling the `Settings.write` method remembers whether or not the "updated version" message has been shown, so it makes sense to only set this once the message actually shows. That way, if someone needs to run with `--quiet`, they can still see the message next time they run `brew update`.

Adding the following line just before the `puts_stdout_or_stderr` calls on lines 192 and 197 should do it:

```ruby
Settings.write "latesttag", new_repository_version
```

**Comment by dkav** [comment]
Amended with @Rylan12 suggestions.

**Review by Rylan12 — APPROVED** [review]
Looks good to me, @dkav! Thanks! 🎉

**Review by MikeMcQuaid — APPROVED** [review]

File: Library/Homebrew/cmd/update-report.rb [hunk 1]
```diff
@@ -91,10 +91,7 @@ def update_report
         "git", "-C", HOMEBREW_REPOSITORY, "tag", "--list", "--sort=-version:refname", "*.*"
       ).lines.first.chomp
 
-      if new_tag != old_tag
-        Settings.write "latesttag", new_tag
-        new_repository_version = new_tag
-      end
+      new_repository_version = new_tag unless new_tag == old_tag
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/update-report.rb:94]:
```suggestion
      new_repository_version = new_tag if new_tag != old_tag
```

**Comment by MikeMcQuaid** [comment]
Thanks so much for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @dkav!

**Comment by carlocab** [comment]
Rebased against master to resolve the failing cask audit.

**Pull request merged**